nuxeo/client.py

```diff
@@ -282,6 +285,20 @@ def request(
 
         return resp
 
+    def _check_headers_and_params_format(self, headers, params):
+        # type: (Dict[Any, Any], Dict[Any, Any]) -> None
+        """Check headers and params keys for dots or underscores and throw a warning if one is found."""
+
+        msg = "{!r} {} should not contain '_' nor '.'. Replace with '-' to get rid of that warning."
+
+        for key in headers:
+            if "_" in key or "." in key:
+                warn(msg.format(key, "header"), DeprecationWarning, 2)
+
+        for key in params:
```

Comment:
A check can be done before that line:
```python
if not isinstance(params, dict):
    return
```

Reply:
I don't really understand the point of doing that. We are already making a check by doing `kwargs.get("params") or {}` before.
If we do the check in the method as you suggest then it means that the type of the `params` var can be  `Dict[text, text]` or `NoneType` instead of just being an empty Dict.

Reply:
The check is not full, what if `params` is a `str` or any object? So the first check if good, and to be sure we are iterating over a `dict`, the second check is required.